### PR TITLE
Adds admin logging to flock stuff

### LINKED ISSUE #10233: Flock has no admin logging
### Labels

[BUG]

### Description

Relay doesn't log, probably other important things don't either

### Steps to reproduce

1. Be an admin
2. The station explodes
3. You don't know why

### Expected Behavior

TODO: Log a bunch of high-impact actions

### Additional Information & Screenshots

_No response_

### PR DESCRIPTION
**Pull request by TobleroneSwordfish**
[BUG] [INTERNAL] [ADMIN]
## About the PR 
Fixes #10233 by adding logging for every flock action I could think of, let me know if I missed any or if some of them don't need to be logged.
Now also adds the new `gamemode` logging type and changes the flock relay and nuclear bomb to use it

## Why's this needed? 
flock devs forgor 💀

### CODE REVIEW

File: code/obj/flock/structure/relay.dm [hunk 1]
```diff
@@ -108,11 +110,12 @@
 		for (var/turf/T as anything in turfs)
 			if (istype(T, /turf/simulated) && !isfeathertile(T))
 				LAGCHECK(LAG_LOW)
-				src?.flock.claimTurf(flock_convert_turf(T))
+				src.flock?.claimTurf(flock_convert_turf(T))
```
Comment (FlameArrow57) [inline code/obj/flock/structure/relay.dm:113]:
```suggestion
				src?.flock?.claimTurf(flock_convert_turf(T))
```

File: code/datums/abilities/flock/flockmind.dm [hunk 2]
```diff
@@ -358,9 +364,10 @@
 		holder.owner.name = "Unknown"
 		R.talk_into(holder.owner, messages, 0, "Unknown")
 		holder.owner.name = name
-	else
+	if (!R)
 		boutput(holder.get_controlling_mob(), "<span class='alert'>That isn't a valid target.</span>")
 		return TRUE
+	logTheThing(LOG_COMBAT, holder.get_controlling_mob(), "casts narrobeam transmission on [constructTarget(R)] with message [message] at [log_loc(src.holder.owner)].")
```
Comment (FlameArrow57) [inline code/datums/abilities/flock/flockmind.dm:370]:
```suggestion
	logTheThing(LOG_COMBAT, holder.get_controlling_mob(), "casts narrowbeam transmission on [constructTarget(R)] with message [message] at [log_loc(src.holder.owner)].")
```
Maybe also who message was sent to?

**Comment by FlameArrow57** [comment]
Would it be better to use the Flockmind or Flocktrace instead of the Flockdrone for abilities? Wasn't sure if it would give client if they are inside Flockdrone, but could maybe report that it was that mob, inside the Flockdrone.

Also, would it be better to report stuff as constructed by the Flockmind, or just the flock the thing belongs to?

File: code/obj/flock/structure/relay.dm (hunk 1 above)
Comment (TobleroneSwordfish) [reply code/obj/flock/structure/relay.dm:113]: I think if `src` is null we probably want it to runtime

Comment (TobleroneSwordfish) [reply code/obj/flock/structure/relay.dm:113]: actually it might be possible, fuck it

**Comment by TobleroneSwordfish** [comment]
Using the flockmind or trace directly results in it not reporting ckey, have switched to reporting flock name instead of flockmind for similar reasons.

**Review by ZeWaka — CHANGES_REQUESTED** [review]
I'd like a new logging category for this - "Gamemode" as a replacement for the Station ones, combat should probably stay combat

**Comment by TobleroneSwordfish** [comment]
> I'd like a new logging category for this - "Gamemode" as a replacement for the Station ones, combat should probably stay combat

I did think it was odd that nuke arming was under station and so forth, will add

**Comment by ZeWaka** [comment]
I did say *new* category, not one replacing the Game category

**Comment by TobleroneSwordfish** [comment]
> I did say _new_ category, not one replacing the Game category

Ah sorry, I thought I'd replace the unused one. Will fix.

**Comment by ZeWaka** [comment]
> Ah sorry, I thought I'd replace the unused one. Will fix.

Isn't unused

**Comment by ZeWaka** [comment]
merge note: needs viewer update, server configs

**Pull request merged**